Applications/AdminUi/apps/admin_ui/lib/home/announcements_overview/modals/create_announcement_dialog.dart

```diff
@@ -0,0 +1,328 @@
+                  final language = announcementTextWidget._languageController.text == ''
+                      ? announcementTextWidget.defaultLanguage
+                      : announcementTextWidget._languageController.text;
+
+                  announcementTexts.add(
+                    AnnouncementText(
+                      title: title,
+                      body: body,
+                      language: language!,
+                    ),
+                  );
+                }
+
+                final response = await GetIt.I.get<AdminApiClient>().announcements.createAnnouncement(
+                  expiresAt: _selectedExpirationDate?.toIso8601String(),
+                  severity: _selectedSeverity!,
+                  announcementTexts: announcementTexts,
+                  recipients: [],
+                );
+
+                if (!context.mounted) return;
+                context.pop();
+
+                if (response.hasError) {
+                  _showErrorSnackbar();
+                  return;
+                }
+                widget.onAnnouncementCreated();
+                _showSuccessSnackbar();
+              }
+            },
+            child: Text(context.l10n.create),
+          ),
+        ),
+      ],
+    );
+  }
+
+  void _remove(_AnnouncementTextFormWidget announcementTextWidget) {
+    final index = _announcementTextWidgets.indexWhere((element) => element == announcementTextWidget);
+
+    setState(() {
+      _announcementTextWidgets.removeAt(index);
+    });
+  }
+
+  void _showSuccessSnackbar() {
+    final snackBar = SnackBar(
+      content: Text(
+        context.l10n.createAnnouncement_announcementSuccess,
+        style: const TextStyle(color: Colors.white),
+      ),
+      backgroundColor: Colors.green,
+      duration: const Duration(seconds: 3),
+      showCloseIcon: true,
+    );
+    ScaffoldMessenger.of(context).showSnackBar(snackBar);
+  }
+
+  void _showErrorSnackbar() {
+    final snackBar = SnackBar(
+      content: Text(
+        context.l10n.createAnnouncement_announcementError,
+        style: const TextStyle(color: Colors.white),
+      ),
+      backgroundColor: Colors.red,
+      duration: const Duration(seconds: 3),
+      showCloseIcon: true,
+    );
+    ScaffoldMessenger.of(context).showSnackBar(snackBar);
+  }
+}
+
+class _AnnouncementTextFormWidget extends StatefulWidget {
+  final GlobalKey<FormState> formKey;
+  final void Function(_AnnouncementTextFormWidget index) onRemove;
+  _AnnouncementTextFormWidget({
+    required this.formKey,
+    required this.onRemove,
+    this.defaultLanguage,
+  });
+
+  @override
+  State<_AnnouncementTextFormWidget> createState() => _AnnouncementTextFormWidgetState();
+
+  final String? defaultLanguage;
+  final TextEditingController _titleController = TextEditingController();
+  final TextEditingController _bodyController = TextEditingController();
+  final TextEditingController _languageController = TextEditingController();
```

Comment:
why aren't these fields on top?

Reply:
The naming with `_` is wrong but I want the .text from the controllers [here](https://github.com/nmshd/backbone/blob/d4e92dae5bba76dae09d42dd585ce8ac5c8daed8/Applications/AdminUi/apps/admin_ui/lib/home/announcements_overview/modals/create_announcement_dialog.dart#L169) so I can't put them in the state or what am I missing?